docs/patterns/cqrs-content.md

```diff
@@ -20,7 +20,7 @@ Read and write workloads are often asymmetrical, with very different performance
 
 CQRS separates reads and writes into different models, using **commands** to update data, and **queries** to read data.
 
-- Commands should be task-based, rather than data centric. ("Book hotel room", not "set ReservationStatus to Reserved").
+- Commands should be task-based, rather than data centric. ("Book hotel room", not "set ReservationStatus to Reserved"). This may require some corresponding changes to the user interaction style. The other part of that is to look at modifying the business logic processing those commands to be successful by default. One technique that supports this is to run some validation rules on the client even before sending the command, possibly disabling buttons, explaining to why on the UI ("no rooms left"). In that manner, the cause for server-side command failures can be narrowed to race conditions (2 users trying to book the last room), and even those can sometimes be addressed with some more data and logic (putting a guest on a waiting list). For more information on these techniques, see the presentation: [Better CQRS through asynchronous user interaction patterns](https://particular.net/videos/cqrs-user-interaction-patterns).
```

Comment:
```suggestion
- Commands should be task-based, rather than data centric. ("Book hotel room", not "set ReservationStatus to Reserved"). This may require some corresponding changes to the user interaction style. The other part of that is to look at modifying the business logic processing those commands to be successful more frequently. One technique that supports this is to run some validation rules on the client even before sending the command, possibly disabling buttons, explaining why on the UI ("no rooms left"). In that manner, the cause for server-side command failures can be narrowed to race conditions (two users trying to book the last room), and even those can sometimes be addressed with some more data and logic (putting a guest on a waiting list).
```